src/Platform/Microsoft.Testing.Platform/Hosts/CommonTestHost.cs

```diff
@@ -61,6 +92,25 @@ public async Task<int> RunAsync()
         return exitCode;
     }
 
+    private async Task<bool> IsDotnetTestHandshakeSuccessfulAsync(NamedPipeClient dotnetTestPipeClient, ITestApplicationModuleInfo? testApplicationModuleInfo, CancellationToken testApplicationCancellationToken)
+    {
+        HandshakeInfo handshakeInfo = new(new Dictionary<string, string>()
+        {
+            { HandshakeInfoPropertyNames.PID, ServiceProvider.GetProcessHandler().GetCurrentProcess().Id.ToString(CultureInfo.InvariantCulture) },
```

Comment:
Getting the process ID can throw if the process is disposed, we had a couple of crashes because of that (e.g. https://github.com/microsoft/testfx/pull/3193)

Reply:
like in the sdk current process cannot be null. I think it's fine here.